### PR TITLE
Support deprecation for constants

### LINKED ISSUE #2544: Add deprecation to module constants
Right now module constants can be annotated with the `@deprecated` annotation but that won't result in a warning if someone tries using a deprecated constant:
```gleam
@deprecated("don't use this, it's evil")
pub const evil_hello = "hello 😈"

pub fn main() {
  io.println(evil_hello)
}
```
Compiles with no warnings

### PR DESCRIPTION
**Pull request by giacomocavalieri**
This PR closes #2544

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## Unreleased
+
+### Language changes
+
+- Warn about deprecated constants
```
Comment (lpil) [inline CHANGELOG.md:7]: Could you write a full sentence here please, mentioning the attribute.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## Unreleased
+
+### Language changes
```
Comment (lpil) [inline CHANGELOG.md:5]: Let's put this as a bug fix

**Review by lpil — COMMENTED** [review]
Thank you! Two small notes inline 🙏

File: CHANGELOG.md (hunk 1 above)
Comment (giacomocavalieri) [reply CHANGELOG.md]: Done!

**Review by lpil — APPROVED** [review]
Thank you once again!!!

**Pull request merged**